libhb: use RtlGetVersion() to get OS version on Windows (MinGW)

marcosfrm opened the pull request:
Microsoft broke GetVersionEx() starting with Windows 8.1. The function now returns incorrect values unless the application manifest specifically lists the newer releases as supported:

https://learn.microsoft.com/en-us/windows/win32/w8cookbook/operating-system-version-changes-in-windows-8-1 https://sourceware.org/pipermail/cygwin/2024-March/255728.html

To avoid this problem, call the documented RtlGetVersion() from ntdll:

https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-rtlgetversion

**Tested on:**

- [x] Windows 10+  (via MinGW)
- [ ] macOS 10.13+
- [ ] Ubuntu Linux

galad87 commented:
Can this be done without dlopen? It looks like we already link with ntdll, so maybe it shouldn't be needed. But if it's the only way, then it's all good.

marcosfrm committed:
libhb: use RtlGetVersion() to get OS version on Windows (MinGW)

Microsoft broke GetVersionEx() starting with Windows 8.1. The function now returns incorrect values unless
the application manifest specifically lists the newer releases as supported:

https://learn.microsoft.com/en-us/windows/win32/w8cookbook/operating-system-version-changes-in-windows-8-1
https://sourceware.org/pipermail/cygwin/2024-March/255728.html

To avoid this problem, call the documented RtlGetVersion() from ntdll:

https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-rtlgetversion

marcosfrm commented:
Thanks for pointing that out. I switched to a direct call to `RtlGetVersion()`. Additionally, I updated the function prototype to match the official Microsoft documentation.

galad87 commented:
Merged, thanks!
